### PR TITLE
Changed vote app to js app

### PR DESCRIPTION
**Pull request by akashkeshari**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,21 +1,395 @@
-    MIT License
```
Comment (shashankbarsin) [inline LICENSE]: Why are we removing this?

File: LICENSE-CODE [hunk 2]
```diff
@@ -0,0 +1,17 @@
+The MIT License (MIT)
```
Comment (shashankbarsin) [inline LICENSE-CODE]: Why 2 separate files called LICENSE and LICENSE-CODE? What's the reason for making license related changes

File: README.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ Create a GitOps configuration referencing this sample repo by using the Azure CL
 After a configuration is created, Azure Arc enabled Kubernetes agents and flux will create use the configuration in the repository to create a Helm release on this cluster.
 
 ```bash
-az k8sconfiguration create --name azure-voting-app \
+az k8sconfiguration create --name azure-js-app \
```
Comment (shashankbarsin) [inline README.md]: Let's call it azure-arc-sample

File: charts/azure-js/Chart.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
 description: A Helm chart for Kubernetes
-name: azure-vote
+name: azure-js
```
Comment (shashankbarsin) [inline charts/azure-js/Chart.yaml]: Let's make it azure-arc-sample

File: charts/azure-js/values.yaml [hunk 5]
```diff
@@ -1,3 +1,3 @@
   
 # Default value for frontendServiceName unless overriden
-frontendServiceName: azure-vote-front
+jsServiceName: azure-js
```
Comment (shashankbarsin) [inline charts/azure-js/values.yaml]: Let's make it azure-sample-app

File: releases/app.yaml [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 kind: HelmRelease
 metadata:
-  name: vote-app
+  name: js-app
   namespace: arc-k8s-demo
 spec:
   releaseName: arc-k8s-demo
   chart:
-    git: https://github.com/Azure/arc-helm-demo
+    git: https://github.com/akashkeshari/pipelines-javascript-docker
```
Comment (shashankbarsin) [inline releases/app.yaml]: After this PR gets merged, the charts will be in https://github.com/Azure/arc-helm-demo repository. So this shouldn't be changed

File: releases/app.yaml [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
     ref: master
-    path: charts/azure-vote
+    path: charts/azure-js
```
Comment (shashankbarsin) [inline releases/app.yaml]: Let's make it azure-arc-sample

File: releases/app.yaml [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
   values:
-    frontendServiceName: arc-k8s-demo-vote-front
\ No newline at end of file
+    frontendServiceName: arc-k8s-demo-js
```
Comment (shashankbarsin) [inline releases/app.yaml]: Let's make it azure-arc-sample or arc-k8s-demo. Drop the js

File: charts/azure-js/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        kubernetes.io/os: linux
+      containers:
+      - name: {{.Release.Name}}-js
+        image: akashkeshari3010/arc-conformance-test:jsappv1
+        ports:
+        - containerPort: 8080
+        resources:
+          requests:
+            memory: "128Mi"
+            cpu: "250m"
```
Comment (shashankbarsin) [inline charts/azure-js/templates/deployment.yaml]: Once we deploy the image on the cluster, we should check how much memory and cpu this is consuming. For requests, let's make the sure the values correspond to the minimum required by the container

File: releases/app.yaml [hunk 10]
```diff
@@ -1,13 +1,13 @@
 apiVersion: helm.fluxcd.io/v1
 kind: HelmRelease
 metadata:
-  name: vote-app
+  name: js-app
```
Comment (shashankbarsin) [inline releases/app.yaml]: azure-arc-sample

File: charts/azure-js/templates/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,35 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Release.Name}}-js
```
Comment (shashankbarsin) [inline charts/azure-js/templates/deployment.yaml]: drop the -js suffix in all occurrences in the file

File: charts/azure-js/templates/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    metadata:
+      labels:
+        app: {{.Release.Name}}-js
+    spec:
+      nodeSelector:
+        kubernetes.io/arch: amd64
+        kubernetes.io/os: linux
+      containers:
+      - name: {{.Release.Name}}-js
+        image: akashkeshari3010/arc-conformance-test:jsappv1
```
Comment (shashankbarsin) [inline charts/azure-js/templates/deployment.yaml]: Let's use the Arc ACR repo here. Pipeline YAML we can check in this PR or the next one, but till then we can consider manually building and pushing the image.

File: charts/azure-js/templates/service.yaml [hunk 13]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Values.jsServiceName }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    name: {{.Release.Name}}-vote-js
```
Comment (shashankbarsin) [inline charts/azure-js/templates/service.yaml]: Release name should suffice. Drop -vote-js

File: charts/azure-js/values.yaml (hunk 5 above)
Comment (shashankbarsin) [inline charts/azure-js/values.yaml]: Let's call it serviceName. And change service name to azure-arc-sample

File: app/package.json [hunk 14]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "my_app",
```
Comment (shashankbarsin) [inline app/package.json:2]: Change this to azure_arc_sample

**Review by shashankbarsin — CHANGES_REQUESTED** [review]

File: LICENSE (hunk 1 above)
Comment (akashkeshari) [reply LICENSE]: added back

File: LICENSE-CODE (hunk 2 above)
Comment (akashkeshari) [reply LICENSE-CODE]: removed LICENSE-CODE

File: README.md (hunk 3 above)
Comment (akashkeshari) [reply README.md]: done

File: charts/azure-js/Chart.yaml (hunk 4 above)
Comment (akashkeshari) [reply charts/azure-js/Chart.yaml]: done

File: charts/azure-js/values.yaml (hunk 5 above)
Comment (akashkeshari) [reply charts/azure-js/values.yaml]: done

File: releases/app.yaml (hunk 6 above)
Comment (akashkeshari) [reply releases/app.yaml]: yep, cheanged back

File: releases/app.yaml (hunk 7 above)
Comment (akashkeshari) [reply releases/app.yaml]: done

File: releases/app.yaml (hunk 8 above)
Comment (akashkeshari) [reply releases/app.yaml]: done

File: charts/azure-js/templates/deployment.yaml (hunk 9 above)
Comment (akashkeshari) [reply charts/azure-js/templates/deployment.yaml]: updated

File: releases/app.yaml (hunk 10 above)
Comment (akashkeshari) [reply releases/app.yaml]: updated

File: charts/azure-js/templates/deployment.yaml (hunk 11 above)
Comment (akashkeshari) [reply charts/azure-js/templates/deployment.yaml]: done

File: charts/azure-js/templates/deployment.yaml (hunk 12 above)
Comment (akashkeshari) [reply charts/azure-js/templates/deployment.yaml]: Yes i will update the image in the pipeline PR

File: charts/azure-js/templates/service.yaml (hunk 13 above)
Comment (akashkeshari) [reply charts/azure-js/templates/service.yaml]: done

File: charts/azure-js/values.yaml (hunk 5 above)
Comment (akashkeshari) [reply charts/azure-js/values.yaml]: done

File: app/package.json (hunk 14 above)
Comment (akashkeshari) [reply app/package.json:2]: done

**Review by shashankbarsin — APPROVED** [review]

File: charts/azure-arc-sample/templates/deployment.yaml [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    metadata:
+      labels:
+        app: {{.Release.Name}}
+    spec:
+      nodeSelector:
+        kubernetes.io/arch: amd64
+        kubernetes.io/os: linux
+      containers:
+      - name: {{.Release.Name}}
+        image: akashkeshari3010/arc-conformance-test:jsappv1
```
Comment (shashankbarsin) [inline charts/azure-arc-sample/templates/deployment.yaml]: Now that we've added pipeline YAML in this PR, we should change this as well. Just push the image once manually to ACR before we merge this PR. When we merge this PR to master branch, configurations already created will try to deploy this chart. Image pull shouldn't fail then

**Review by shashankbarsin — CHANGES_REQUESTED** [review]

**Review by shashankbarsin — APPROVED** [review]

**Pull request merged**